File: MotorTherapyGame/Assets/Scripts/Game.cs
using System;
using UnityEngine;

/**
 * @Author Jose Acuna
 * Last time edited 23/10/19 by Jose Acuna
 * @Reference https://docs.unity3d.com/ScriptReference/JsonUtility.FromJson.html
 */
[Serializable]
public class Game
{
    public string type;
    public string status;
    public Piano piano;
    public Targets targets;
    public CobWeb cobWeb;
    public Balloons balloons;

    public Game() {}
    
    
    public Game(string type, string status)
    {
        this.type = type;
        this.status = status;
    }
    
    
    public static Game CreateFromJson(string json)
    {
        return JsonUtility.FromJson<Game>(json);
    }
    
    
    [Serializable]
    public struct Piano
    {
        public string[] colors;
        public int[] points;
        public int time;
    }
    
    [Serializable]
    public struct Targets
    {
        public int x;
        public int y;
        public int time;
    }
    
    [Serializable]
    public struct CobWeb
    {
        public Card[] cards;
        public int rows;
        public int columns;
        
        [Serializable]
        public struct Card
        {
            public string name;
            public int points;
            public int i;
            public int j;
        }
    }
    
    [Serializable]
    public struct Balloons
    {
        public int x;
        public int y;
    }

    public CobWeb.Card GetCard(int i, int j)
    {
        var result = new CobWeb.Card();
        foreach (var card in cobWeb.cards)
        {
            if (card.i != i || card.j != j) continue;
            result = card;
            break;
        }

        return result;
    }
}

File: Compiler/src/game/Game.py
# @Author Jose Daniel Acuna
# Last time edited 11/1/19
# Reference https://www.journaldev.com/16016/python-parse-json-dumps-loads

import json


# Class for Json encoding
from collections import namedtuple


class Game:

    def __init__(self, type="", status="", piano=None, targets=None, cobWeb=None, balloons=None):
        self.type = type
        self.status = status
        self.piano = piano
        self.targets = targets
        self.cobWeb = cobWeb
        self.balloons = balloons

    def dict(self):
        return {
            'type': self.type,
            'status': self.status,
            'piano': self.piano,
            'targets': self.targets,
            'cobWeb': self.cobWeb,
            'balloons': self.balloons}

    def to_json(self):
        return json.dumps(self.dict())

    @staticmethod
    def from_json(json_str):
        dict = json.loads(json_str)
        return namedtuple("Game", dict.keys())(*dict.values())
